internal/build/job/manager.go

```diff
@@ -83,6 +87,20 @@ func (jbm *jobManager) Sync(ctx context.Context, mod kmmv1beta1.Module, m kmmv1b
 		return build.Result{Status: build.StatusCreated, Requeue: true}, nil
 	}
 
+	changed, err := jbm.isJobChanged(mod, buildConfig, targetKernel, m.ContainerImage, pushImage, job)
+	if err != nil {
+		return build.Result{}, fmt.Errorf("could not determine if job has changed: %v", err)
+	}
+
+	if changed {
+		logger.Info("Job spec changed, deleting..", "name", job.Name)
+		err = jbm.client.Delete(ctx, job)
+		if err != nil {
+			logger.Info(utils.WarnString(fmt.Sprintf("failed to delete build job %s", job.Name)))
```

Comment:
```suggestion
		if err := jbm.client.Delete(ctx, job); err != nil {
			logger.Info(utils.WarnString(fmt.Sprintf("failed to delete build job %s: %v", job.Name, err)))
```

Reply:
Could we patch the Job instead of deleting it?

Reply:
no, jobs specs cannot not be patched, kubernetes restriction